lz4/stream/_stream.c

```diff
@@ -0,0 +1,1404 @@
+   *          = (((csize - 16) * 256) - (csize - 16)) / 256
+   *          = ((csize * 256) - (16 * 256) - (csize - 16)) / 256
+   *          = ((csize << 8) - (16 << 8) - csize + 16) >> 8
+   *          = ((csize << 8) - csize + 16 - (16 << 8)) >> 8
+   *          = ((csize << 8) - csize - 4080) >> 8
+   *
+   *  Notes:
+   *  - Using 64-bit long integer for intermediate computation to avoid any
+   *    truncation when shifting left large csize values.
+   *  - Due to the round integer approximation, running the following
+   *    calculation can give a non-null result:
+   *      result = n - _LZ4_inputBound( _LZ4_compressBound( n ) )
+   *    but in all cases, this difference is between 0 and 1.
+   *    Thus, the valid maximal input size returned by this funtcion is
+   *    incremented by 1 to avoid any buffer overflow in case of decompression
+   *    in a dynamically allocated buffer.
+   *  - For small compressed length (shorter than 16 bytes), make sure a
+   *    non-null size is returned.
+   */
+  if (csize < 16)
+    {
+      csize = 17;
+    }
+
+  if (csize <= LZ4_compressBound(LZ4_MAX_INPUT_SIZE))
+    {
+      isize = ((csize << 8) - csize - 4080) >> 8;
+
+      if (isize > (uint32_t)LZ4_MAX_INPUT_SIZE)
+        {
+          isize = 0;
+        }
+      else
+        {
+          isize += 1;
+        }
+    }
+
+  return (uint32_t)(isize & UINT32_MAX);
+}
+
+
+#ifdef DUMP_CONTEXT
+static void
+_LZ4StreamContext_dump (const _LZ4StreamContext_t * lz4_ctxt)
+{
+#define __str(s)  # s
+#define _str(s)   __str(s)
+#define dump_attr(ptr, path, fmt) \
+  printf("  %s->%-30s = " fmt "\n", _str(ptr), _str(path), (ptr)->path);
+
+  printf("  %s%-32s = " "%p" "\n", _str(lz4_ctxt), "", lz4_ctxt);
+
+  if (lz4_ctxt == NULL)
+    {
+      return;
+    }
+
+  dump_attr (lz4_ctxt, double_buffer.buf, "%p");
+  dump_attr (lz4_ctxt, double_buffer.page_size, "%u");
+  for (int i = DOUBLE_BUFFER_INDEX_MIN;
+       i < (DOUBLE_BUFFER_INDEX_MIN + DOUBLE_BUFFER_PAGE_COUNT);
+       ++i)
+    {
+      printf ("  %s->%s[%d] = " "%p" "\n",
+              _str (lz4_ctxt), _str (double_buffer.pages), i,
+              lz4_ctxt->double_buffer.pages[i]);
+    }
+  dump_attr (lz4_ctxt, double_buffer.index, "%d");
+
+  dump_attr (lz4_ctxt, output.buf, "%p");
+  dump_attr (lz4_ctxt, output.len, "%u");
+
+  dump_attr (lz4_ctxt, lz4_state.compress.fast, "%p");
+  dump_attr (lz4_ctxt, lz4_state.compress.hc, "%p");
+  dump_attr (lz4_ctxt, lz4_state.decompress, "%p");
+
+  dump_attr (lz4_ctxt, config.acceleration, "%d");
+  dump_attr (lz4_ctxt, config.compression_level, "%d");
+  dump_attr (lz4_ctxt, config.store_comp_size, "%d");
+  dump_attr (lz4_ctxt, config.return_bytearray, "%d");
+  dump_attr (lz4_ctxt, config.direction, "%d (0: compress / 1: decompress)");
+  dump_attr (lz4_ctxt, config.comp, "%d");
+
+#undef dump_attr
+#undef _str
+#undef __str
+}
+#else /* DUMP_CONTEXT */
+#define _LZ4StreamContext_dump(...)
+#endif /* DUMP_CONTEXT */
+
```

Comment:
This all looks like debug code you used during development _ I really don't think we should clutter up the code with this. You could either delete it, or move it to a separate file that can be simply included for debugging purposes, if it is useful for the future.

Reply:
ok, I'll remove it